lib/irb/init.rb

```diff
@@ -44,7 +44,7 @@ def IRB.init_config(ap_path)
     @CONF[:IRB_RC] = nil
 
     @CONF[:USE_SINGLELINE] = false unless defined?(ReadlineInputMethod)
-    @CONF[:USE_COLORIZE] = !ENV['NO_COLOR']
+    @CONF[:USE_COLORIZE] = !ENV['NO_COLOR']&.slice(0)
```

Comment:
This is a brilliant way to check emptiness! But it took me a while to comprehend because of usage of `!` + `slice`. 

So personally I still prefer the common idioms like

```suggestion
    @CONF[:USE_COLORIZE] = (nc = ENV['NO_COLOR']).nil? || nc.empty?
```

Reply:
That condition is inverted.